### PR TITLE
Add cpplint pre-commit hook

### PR DESCRIPTION
**Pull request by wangkuiyi**
We have many code style problems in even the most recent Fluid code. An example is https://github.com/PaddlePaddle/Paddle/issues/9505.  I am worried that during the period of performance profiling and tuning, which implies very high pressure to the team, we'd care even less about code style. So I am add cpplint into our pre-commit hook.  

Fixes https://github.com/PaddlePaddle/Paddle/issues/9510

### CODE REVIEW

**Comment by wangkuiyi** [comment]
I also disabled dropout_op_test in this PR (https://github.com/PaddlePaddle/Paddle/issues/9507)

**Comment by typhoonzero** [comment]
Thought we already have clang-format, and in `.clang-format` it uses google c++ style to format the code.

**Comment by gongweibao** [comment]
It seems that `clang-format` use a subset of Google-style and it only does a format and doesn't do a semantic transformation.
Such as `(int)a => static_cast<int>(a)`.

**Comment by wangkuiyi** [comment]
clang-format can but can only fix a small fraction of cpplint errors. @typhoonzero

File: tools/codestyle/cpplint_pre_commit.hook [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+TOTAL_ERRORS=0
+
+# The trick to remove deleted files: https://stackoverflow.com/a/2413151
+for file in $(git diff --cached --name-status | awk '$1 != "D" {print $2}'); do
```
Comment (helinwang) [inline tools/codestyle/cpplint_pre_commit.hook]: So CI won't check cpplint since git diff is clean when running CI.

**Pull request merged**